apps/main_app/lib/main_app.ex

```diff
@@ -3,19 +3,25 @@ defmodule MainApp do
 
   use Application
 
-  @subscriber Application.compile_env(:main_app, :subscriber, nil)
   @replicator Application.compile_env(:main_app, :replicator, nil)
-  @publisher Application.compile_env(:main_app, :publisher, nil)
+  @publishers Application.compile_env(:main_app, :publishers, [])
 
   @impl true
   def start(_type, _args) do
     children = [
-      @subscriber,
+      MainApp.Communicator,
+      Enum.map(@publishers, fn %{name: name, init_arg: init_arg, module: module} ->
+        %{
+          id: name,
+          start: {module, :start_link, [name, init_arg]}
+        }
+      end),
       {@replicator,
-       [host: "localhost", database: "postgres", username: "postgres", password: "postgres"]},
-      {@publisher, "/tmp/replication.log"},
+       [host: "localhost", database: "postgres", username: "postgres", password: "postgres"]}
     ]
 
+    children = List.flatten(children)
```

Comment:
I suggest flattening publishers only. Maybe even outside of (before) `children = [...]`?